documentation/modules/configuring/con-config-mirrormaker2-high-volume-messages.adoc

```diff
@@ -0,0 +1,70 @@
+// Module included in the following assemblies:
+//
+// assembly-config-mirrormaker2.adoc
+
+[id='con-mirrormaker-high-volume-messages-{context}']
+= Handling high volumes of messages
+
+[role="_abstract"]
+If your MirrorMaker 2.0 deployment is going to be handling a high volume of messages, you might need to adjust its configuration to support it.
+
+The flush pipeline for data replication is *source topic -> (Kafka Connect) source message queue -> producer buffer -> target topic*.
+An offset flush timeout period (`offset.flush.timeout.ms`) is the time to wait for the producer buffer (`producer.buffer.memory`) to flush and offset data to be committed.
+Try to avoid a situation where a large producer buffer and an insufficient offset flush timeout period causes a _failed to flush_ type of error.
```

Comment:
We actually have 1 common cause, but 2 different error messages: 
- "Failed to flush" is related to `producer.buffer.memory`. When this buffer is full, the producer's `send()` operation becomes synchronous and it slows down up to the point it hits timeout. 
- "Failed to commit offsets" is related to `offset.flush.timeout.ms`. KafkaConnect commits the offset in the source cluster periodically (every 1 minute by default), but it may hit timeout when there are too many messages during this interval.